### DISCUSSION TITLE
Notice T02 field OPP-032-Tender

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#420] [Q&A] [answered]
Hi Lab !
making T02 xml notice I get an xsd error  in one section efac:ContractTerm : 
<efac:LotTender>
							<cbc:ID schemeName="tender">TEN-0001</cbc:ID>
							**<efac:ContractTerm>
								<efbc:TermPercent>400</efbc:TermPercent>
							</efac:ContractTerm>**
							<efac:ContractTerm>
								<efbc:TermCode listName="contract-detail">cost-comp</efbc:TermCode>
								<efbc:TermDescription languageID="ENG">90</efbc:TermDescription>
							</efac:ContractTerm>
							<efac:ContractTerm>
								<efbc:TermCode listName="rewards-penalties">rew-pen</efbc:TermCode>
								  <efbc:TermDescription languageID="ENG">IN CASTIGO</efbc:TermDescription>
							</efac:ContractTerm>
						</efac:LotTender>

efbc:TermPercent is field OPP-032-Tender, while efac:ContractTerm> is ND-RevenueAllocation , ND-RevenueAllocation has got only one 'son' that is OPP-032-Tender. 
ND-RevenueAllocation's xpathrelative is "efac:ContractTerm[efbc:TermCode/text()='all-rev-tic']" and the efbc:TermCode is what it lacks in xsd but is in the predicate
Can you please explain how making the xml so that it is xsd valid ? 
Thanks a lot
Renato

**Comment by rouschr** [comment, marked as answer]
Hello,
are you relying on the notice editor demo?
Are you using SDK 1.6?
In the application.yaml of the editor demo you have:
```
versions: 1.0,1.1,1.2,1.3,1.4,1.5 #,1.6
```
You have to uncomment 1.6:
```
versions: 1.0,1.1,1.2,1.3,1.4,1.5,1.6
```
This is because 1.6 was released after this work was done in the editor demo.
The big difference between 1.5 and 1.6 is here:
https://github.com/OP-TED/eForms-SDK/blob/1.6.0/notice-types/notice-types.json

**Reply by rouschr** [reply]
Note that now in the _develop_ branch 1.6 is uncommented.

**Reply by rengit62** [reply]
Yes I am using SDK 1.6,  but I am not relying on the notice editor.
In the form that I Attach , Revenues Allocation, Contract conditions  code, Contract conditions description (), Penalties and Rewards code and Description are all mandatory, although in xsd it is mandatory only TermCode is mandatory : 
<xsd:complexType name="ContractTermType">
		<xsd:sequence>
			<xsd:element ref="efbc:TermCode" minOccurs="1" maxOccurs="1"/>
			<xsd:element ref="efbc:TermDescription" minOccurs="0" maxOccurs="unbounded"/>
			<xsd:element ref="efbc:TermIndicator" minOccurs="0" maxOccurs="1"/>
			<xsd:element ref="efbc:TermPercent" minOccurs="0" maxOccurs="1"/>
		</xsd:sequence>
	</xsd:complexType>

But looking at my error : 
<efac:LotTender>
							<cbc:ID schemeName="tender">TEN-0001</cbc:ID>
							<efac:ContractTerm>
								<efbc:TermPercent>80</efbc:TermPercent>
							</efac:ContractTerm>
							<efac:ContractTerm>
								<efbc:TermCode listName="contract-detail">all-rev-tic</efbc:TermCode>
								<efbc:TermDescription languageID="ENG">LA DONZELLETTA</efbc:TermDescription>
							</efac:ContractTerm>
							<efac:ContractTerm>
								<efbc:TermCode listName="rewards-penalties">rew-pen</efbc:TermCode>
								<efbc:TermDescription languageID="ENG">VIEN DALLA CAMPAGNA</efbc:TermDescription>
							</efac:ContractTerm>
						</efac:LotTender>

<efbc:TermPercent> should be inside a tag where is at least efbc:TermCode but watching the definition in fields.json it seams that  ND-RevenueAllocation and ND-OtherContractExecutionConditions can be alternative  : 
 **"id" : "ND-RevenueAllocation",**
    "parentId" : "ND-LotTender",
    "xpathAbsolute" : "/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:LotTender/efac:ContractTerm[efbc:TermCode/text()='all-rev-tic']",
    **"xpathRelative" : "efac:ContractTerm[efbc:TermCode/text()='all-rev-tic']",**
    "repeatable" : false

**"id" : "ND-OtherContractExecutionConditions"**,
    "parentId" : "ND-LotTender",
    "xpathAbsolute" : "/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:LotTender/efac:ContractTerm[not(efbc:TermCode/text()='all-rev-tic')][efbc:TermCode/@listName='contract-detail']",
    **"xpathRelative" : "efac:ContractTerm[not(efbc:TermCode/text()='all-rev-tic')][efbc:TermCode/@listName='contract-detail']",**
    "repeatable" : true
  },

Can you confirm this one ?

Last question about T02 notice is the sample notice T02_ESP.xml  about : 
<efac:ContractTerm>
								<efbc:TermCode listName="contract-detail">cost-comp</efbc:TermCode>
								<efbc:TermDescription languageID="SPA">Description of the costs parameters blablabla ...</efbc:TermDescription>
								<efbc:TermIndicator>true</efbc:TermIndicator>
							</efac:ContractTerm>
I did not find efbc:TermIndicator in fields.json	
Thanks and Regards
Renato						

![T02Error](https://user-images.githubusercontent.com/52211037/235678102-c9a9d2c3-91a1-4bf6-bcd2-0e2a00343702.jpg)

**Reply by YvesJo** [reply]
efbc:TermIndicator is not needed as the description shall be specified when the indicator is set to true and the description shall not exist when the indicator is set to false. This indicator will also be removed from the XSD in a future release.
